server/model/monitor.js

```diff
@@ -107,7 +107,7 @@ class Monitor extends BeanModel {
             pathName,
             parent: this.parent,
             childrenIDs: await Monitor.getAllChildrenIDs(this.id),
-            url: this.url,
+            url: this.getUrl()?.href,
```

Comment:
lets not change this one.
As stated previously, I have not looked into the other notification providers/monitors and don't think that this issue needs to do this.
I am thinking about notification providers such as webhooks allowing templating or another monitor using this field for somethign which is not considered a url by node.

Without good looking into, this is too risky for me.

```suggestion
            url: this.url,
```